catalog/spatial/geocoding/spatial-geocoding-lucene/src/main/java/org/codice/ddf/spatial/geocoding/index/GeoNamesCatalogIndexer.java

```diff
@@ -0,0 +1,282 @@
+        new GeoEntryExtractor.ExtractionCallback() {
+          @Override
+          public void extracted(final GeoEntry newEntry) throws GeoEntryIndexingException {
+            Metacard metacard = transformGeoEntryToMetacard(newEntry);
+            if (metacard != null) {
+              metacardList.add(metacard);
+            }
+          }
+
+          @Override
+          public void updateProgress(final int progress) {
+            if (progressCallback != null) {
+              progressCallback.updateProgress(progress);
+            }
+          }
+        };
+
+    if (create) {
+      removeGeoNamesMetacardsFromCatalog(extractionCallback);
+    }
+
+    geoEntryExtractor.pushGeoEntriesToExtractionCallback(resource, extractionCallback);
+
+    if (CollectionUtils.isEmpty(metacardList)) {
+      LOGGER.debug("No Metacards were created from the resource.");
+      return;
+    }
+
+    executeCreateMetacardRequest(metacardList);
+  }
+
+  private void removeGeoNamesMetacardsFromCatalog(ProgressCallback extractionCallback) {
+    extractionCallback.updateProgress(0);
+
+    Optional<CatalogProvider> catalogProviderOptional = catalogProviders.stream().findFirst();
+    if (catalogProviderOptional.isPresent()) {
+      CatalogProvider catalogProvider = catalogProviderOptional.get();
+
+      Query query =
+          new QueryImpl(
+              filter, 1, BATCH_SIZE, SortBy.NATURAL_ORDER, false, TimeUnit.SECONDS.toMillis(10));
+      QueryRequest queryRequest = new QueryRequestImpl(query);
+
+      List<Serializable> metacardsToDelete;
+      try {
+        metacardsToDelete =
+            ResultIterable.resultIterable(catalogFramework::query, queryRequest)
+                .stream()
+                .map(Result::getMetacard)
+                .map(Metacard::getId)
+                .collect(Collectors.toList());
+      } catch (Exception e) {
+        LOGGER.debug("Unable to query for metacards.", e);
+        return;
+      }
+
+      if (CollectionUtils.isEmpty(metacardsToDelete)) {
+        LOGGER.debug("No metacards to delete.");
+        return;
+      }
+
+      for (int i = 0; i < metacardsToDelete.size(); i += BATCH_SIZE) {
+
+        int lastIndex = i + BATCH_SIZE;
+        if (lastIndex > metacardsToDelete.size()) {
+          lastIndex = metacardsToDelete.size();
+        }
+
+        List<Serializable> sublist = metacardsToDelete.subList(i, lastIndex);
+
+        DeleteRequest deleteRequest = new DeleteRequestImpl(sublist, Core.ID, new HashMap<>());
+        try {
+          DeleteResponse deleteResponse = catalogProvider.delete(deleteRequest);
+          List<Metacard> deletedMetacards = deleteResponse.getDeletedMetacards();
+          LOGGER.debug(
+              "{} metacards deleted.", deletedMetacards == null ? 0 : deletedMetacards.size());
+          extractionCallback.updateProgress(
+              (int) (((double) (i + 1) / metacardsToDelete.size()) * 50));
```

Comment:
what does `50` mean?

Reply:
I think, with this implementation, the delete operations only take up the first 50% of the overall progress in the Karaf console

Reply:
Yeah, this is just for the progress callback.  What would an appropriate variable name for it even be?

Reply:
I think that in the context of the `GazetteerUpdateCommand`, the "delete old gazetteer data" (if, say, `-c` is used) takes up the first 50% of the progress and the "insert new gazetteer data" portion takes up the other 50%